### PR TITLE
[RLlib] Cherry pick rllib test fixes 2 6

### LINKED ISSUE #37039: Release test rllib_learner_group_checkpointing_multinode.aws failed
Release test **rllib_learner_group_checkpointing_multinode.aws** failed. See https://buildkite.com/ray-project/release-tests-branch/builds/1857#01891d64-3ff1-456f-92cb-ea5ecf6adb3e for more details.

Managed by OSS Test Policy

### LINKED ISSUE #37078: Release test rllib_learning_tests_dqn_tf.aws failed
Release test **rllib_learning_tests_dqn_tf.aws** failed. See https://buildkite.com/ray-project/release-tests-branch/builds/1861#01892162-da0a-4381-9445-ab0a708e046c for more details.

Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by avnishn**
fixes for #37078 and #37039

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/rllib_tests/multi_node_checkpointing_compute_config.yaml [hunk 1]
```diff
@@ -9,9 +9,9 @@ head_node_type:
 
 worker_node_types:
     - name: worker_node
-      instance_type: g3.8xlarge
-      min_workers: 1
-      max_workers: 1
+      instance_type: g3s.xlarge
+      min_workers: 2
+      max_workers: 2
```
Comment (can-anyscale) [inline release/rllib_tests/multi_node_checkpointing_compute_config.yaml:14]: oh hmm if you change the number of workers here, you need to change the wait_for_node in release_tests.yaml as well

**Comment by can-anyscale** [comment]
Do you need to change the wait_for_node in release_tests.yaml

File: release/rllib_tests/multi_node_checkpointing_compute_config.yaml (hunk 1 above)
Comment (avnishn) [reply release/rllib_tests/multi_node_checkpointing_compute_config.yaml:14]: oh shucks. It seems to have all worked out though?

Comment (avnishn) [reply release/rllib_tests/multi_node_checkpointing_compute_config.yaml:14]: tests were previously passing.

Comment (can-anyscale) [reply release/rllib_tests/multi_node_checkpointing_compute_config.yaml:14]: yes, in theory it will be a data race bug so it might or might not happen

**Pull request merged**